pkg/controller/cluster/main-controller.go

```diff
@@ -516,8 +516,82 @@ func (c *Controller) syncHandler(key string) error {
 			c.recorder.Event(mi, corev1.EventTypeWarning, ErrResourceExists, msg)
 			return fmt.Errorf(msg)
 		}
+
 		// keep track of all replicas
 		totalReplicas += ss.Status.Replicas
+		images = append(images, ss.Spec.Template.Spec.Containers[0].Image)
+	}
+
+	var sameImageCount int
+	for _, image := range images {
+		if mi.Spec.Image == image {
```

Comment:
This condition will always fail when there is a new image update requested, sent a PR https://github.com/harshavardhana/minio-operator/pull/1 PTAL